hm_1/tasks/task_01_04.py

```diff
@@ -0,0 +1,14 @@
+from typing import List
+
+
+def check_sum_of_four(a: List[int], b: List[int], c: List[int], d: List[int]) -> int:
+    count = 0
+    [
+        count := count + 1
+        for i in d
+        for j in c
+        for k in b
+        for h in a
+        if i + j + k + h == 0
+    ]
```

Comment:
What is the complexity of this code?

Reply:
As I understood, I think it's O(n^4), because in the task's description was written that lengths of lists are equal and their size is "n". It's must be right if the complexity of list comprehension has the same as an ordinary loop. I read that is so.

Reply:
This is correct, Let's try to simplify it to improve it to O^3 using other default collection.